### PR TITLE
UIOR-495 fix poNumber suffix and prefix usage

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIOR-495

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Utils/orderResource.js [hunk 1]
```diff
@@ -22,6 +22,14 @@ export const getFullOrderNumber = ({ poNumberPrefix = '', poNumberSuffix = '', p
 export const updateOrderResource = (order, mutator, changedProps) => {
   const clonedOrder = cloneDeep(order);
 
+  Object.assign(clonedOrder, changedProps);
+
+  return saveOrder(clonedOrder, mutator);
+};
+
+export const editOrderResource = (order, mutator, changedProps) => {
```
Comment (aliaksei-chumakou) [inline src/components/Utils/orderResource.js:30]: it's totally similar names , editOrderResource  and updateOrderResource, so no clear purpose of introducing another similar method

Comment (AndreiShumski2) [reply src/components/Utils/orderResource.js:30]: right. 
- `editOrderResource` is for editing PO (in this case PO number is parsed  to purePoNumber - numbers only with `initialValues` for form. And before saving PONumber = poNumberPrefix+poNumber+poNumberSuffix);
- `updateOrderResource` for all other cases such as open, approve, delete order - where no needed to change poNumber.

Comment (aliaksei-chumakou) [reply src/components/Utils/orderResource.js:30]: Is that a problem to use updateOrderResource ? IMHO, idea of that function was to be a service function to call API.

File: src/components/LayerCollection/LayerPO.js [hunk 2]
```diff
@@ -90,13 +90,12 @@ function LayerPO({
...
 
   const updatePO = useCallback(values => {
+    const poNumber = getFullOrderNumber(values) || undefined;
+
     setIsLoading(true);
-    setSavingValues(values);
-    const saveFn = values.id
-      ? updateOrderResource
-      : createOrderResource;
+    setSavingValues({ ...values, poNumber });
```
Comment (aliaksei-chumakou) [inline src/components/LayerCollection/LayerPO.js:96]: but that means in case of error `poNumber` will be filled with prefix and suffix. no?

Comment (aliaksei-chumakou) [reply src/components/LayerCollection/LayerPO.js:96]: IMHO, you should keep `createOrderResource` and maybe it makes sense to call it `createOrEditOrderResource`, also change name of its first param from `order` to `orderFormValues` might help for readability

Comment (AndreiShumski2) [reply src/components/LayerCollection/LayerPO.js]: fixed

File: src/components/Utils/orderResource.js [hunk 3]
```diff
@@ -22,17 +22,15 @@ export const getFullOrderNumber = ({ poNumberPrefix = '', poNumberSuffix = '', p
...
 
   return saveOrder(clonedOrder, mutator);
 };
 
-export const createOrderResource = (order, mutator) => {
-  const clonedOrder = cloneDeep(order);
+export const createOrEditOrderResource = (orderFormValues, mutator, changedProps) => {
+  const clonedOrder = cloneDeep(orderFormValues);
 
-  clonedOrder.poNumber = getFullOrderNumber(clonedOrder) || undefined;
```
Comment (aliaksei-chumakou) [inline src/components/Utils/orderResource.js:35]: what is the point of moving this into Component?

Comment (AndreiShumski2) [reply src/components/Utils/orderResource.js:35]: misunderstood your previous comment

**Pull request merged**